pcicrawler/ocp_diag.py

```diff
@@ -0,0 +1,259 @@
+                    missing_devices.append("No device was found with address {}.".format(identifier['address']))
+                potential_duts = [dut for dut in potential_duts if str(dut) == identifier['address']]
+            if 'vendor_id' in identifier:
+                potential_duts = [dut for dut in potential_duts if dut.vendor_id == identifier['vendor_id']]
+                if len(potential_duts) == 0:
+                    missing_devices.append("No device was found with vendor id {}.".format(identifier['vendor_id']))
+            if 'device_id' in identifier:
+                potential_duts = [dut for dut in potential_duts if dut.device_id == identifier['device_id']]
+                if len(potential_duts) == 0:
+                    missing_devices.append("No device was found with device id {}.".format(identifier['device_id']))
+            if len(potential_duts) != 0:
+                filtered_duts.append(TestSet(potential_duts, dut['validate']))
+
+        # Create the dut information for the test run
+        dut = tv.Dut(id="0", name=socket.gethostname())
+        devices_seen = []
+        for test_set in filtered_duts:
+            for pcie_device in test_set.duts:
+                if pcie_device not in devices_seen:
+                    devices_seen.append(pcie_device)
+                    dut.add_hardware_info(name=pcie_device.name,
+                                          location=str(pcie_device.location),
+                                          manufacturer=str(pcie_device.vendor_id),
+                                          )
+        dut.add_software_info(name="pcicrawler",
+                              revision=pkg_resources.get_distribution("pcicrawler").version)
+
+        # Start checks
+        with self._ocp_run.scope(dut=dut):
+            # First we can list out the missing devices
+            if len(missing_devices) > 0:
+                self._result = False
+                for missing_device_msg in missing_devices:
+                    self._ocp_run.add_log(tv.objects.LogSeverity.ERROR, missing_device_msg)
+            for test_set in filtered_duts:
+                # Run checks for each slot
+                for dut in test_set.duts:
+                    # Run available tests
+                    for test_name in AVAILABLE_TESTS:
+                        if test_name in test_set.conditions:
+                            conditions = test_set.conditions[test_name]
+                            if test_name == 'check_AER':
+                                conditions = None
+                            test = AVAILABLE_TESTS[test_name](
+                                dut,
+                                conditions,
+                                self._ocp_run
+                            )
+                            result = test.run()
+                            if not self._result and not result:
+                                self._result = result
```

Comment:
so if `self._result == False` and `result == False`, set `self._result = result` ? what does this accomplish
maybe you wanted
```
if not result:
    self._result = False
```

Reply:
i'm just avoiding reassigning the value of self._result if I already have a false value in the final result placeholder.

Reply:
ok but why? this looks overly complicated just to avoid an assignment op?

Reply:
if I recall it's because once the value was set I didn't want it to change.